cmd/dp-filter-api/main.go

```diff
@@ -101,79 +113,125 @@ func main() {
 		cfg.DownloadServiceSecretKey,
+		cfg.ServiceAuthToken,
 		auditor,
+		&hc,
 	)
 
-	go func() {
-		select {
-		case err := <-producer.Errors():
-			log.ErrorC("kafka producer error received", err, nil)
-		case err := <-auditProducer.Errors():
-			log.ErrorC("kafka audit producer error received", err, nil)
-		case err := <-apiErrors:
-			log.ErrorC("api error received", err, nil)
-		}
-	}()
-
 	// block until a fatal error occurs
 	select {
 	case <-signals:
-		log.Info("os signal received", nil)
+		log.Event(ctx, "os signal received")
 	}
 
-	log.Info(fmt.Sprintf("Shutdown with timeout: %s", cfg.ShutdownTimeout), nil)
+	log.Event(ctx, fmt.Sprintf("Shutdown with timeout: %s", cfg.ShutdownTimeout), log.INFO)
 	ctx, cancel := context.WithTimeout(context.Background(), cfg.ShutdownTimeout)
 
 	// Gracefully shutdown the application closing any open resources.
 	go func() {
 		defer cancel()
 
 		if err = api.Close(ctx); err != nil {
-			logIfError(err, "unable to close api server")
+			logIfError(ctx, err, "unable to close api server")
 		}
 
-		healthTicker.Close()
+		hc.Stop()
 
 		if serviceList.FilterStore {
-			log.Info("closing filter store", nil)
+			log.Event(ctx, "closing filter store")
 			// mongo.Close() may use all remaining time in the context
-			logIfError(mongolib.Close(ctx, dataStore.Session), "unable to close filter store")
+			logIfError(ctx, mongolib.Close(ctx, dataStore.Session), "unable to close filter store")
 		}
 
 		if serviceList.ObservationStore {
-			log.Info("closing observation store", nil)
-			logIfError(observationStore.Close(ctx), "unable to close observation store")
+			log.Event(ctx, "closing observation store")
+			logIfError(ctx, observationStore.Close(ctx), "unable to close observation store")
 		}
 
 		if serviceList.FilterOutputSubmittedProducer {
-			log.Info("closing filter output submitted producer", nil)
+			log.Event(ctx, "closing filter output submitted producer")
 			// Close producer after http server has closed so if a message
 			// needs to be sent to kafka off a request it can
-			logIfError(producer.Close(ctx), "unable to close filter output submitted producer")
+			logIfError(ctx, producer.Close(ctx), "unable to close filter output submitted producer")
 		}
 
 		if serviceList.AuditProducer {
-			log.Info("closing audit producer", nil)
-			logIfError(auditProducer.Close(ctx), "unable to close audit producer")
+			log.Event(ctx, "closing audit producer")
+			logIfError(ctx, auditProducer.Close(ctx), "unable to close audit producer")
 		}
 	}()
 
 	// wait for shutdown success (via cancel) or failure (timeout)
 	<-ctx.Done()
 
-	log.Info("Shutdown complete", nil)
+	log.Event(ctx, "Shutdown complete")
 	os.Exit(1)
 }
 
-func exitIfError(err error, message string) {
+func startHealthCheck(ctx context.Context, cfg *config.Config, datasetAPI *dataset.Client, producer *kafka.Producer, observationStore *graph.DB, dataStore *mongo.FilterStore, auditProducer *kafka.Producer) healthcheck.HealthCheck {
```

Comment:
I like this approach of having the creation and register checkers in a function outside main func :)